libraries/dotnet/EnergyOrigin.ActivityLog/EnergyOrigin.ActivityLog/HostedService/CleanupActivityLogsHostedService.cs

```diff
@@ -0,0 +1,48 @@
+using EnergyOrigin.ActivityLog.DataContext;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace EnergyOrigin.ActivityLog.HostedService;
+
+public class CleanupActivityLogsHostedService(
+    ILogger<CleanupActivityLogsHostedService> logger,
+    IServiceProvider services, IOptions<ActivityLogOptions> activityLogOptions)
+    : BackgroundService
+{
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        logger.LogInformation($"{nameof(CleanupActivityLogsHostedService)} running.");
+
+        // On startup, delete all activity logs older than 6 months
+        await DeleteActivityLogsOlderThan6Months();
```

Comment:
Having it in the while loop only is ok I'd say and do something like:
while (!stoppingToken.IsCancellationRequested)
{
    //Delete logs
    await Sleep(stoppingToken);
}